### PR TITLE
[TTAHUB-2150] Sorting sessions

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

This changes the session sort in 3 locations.

1. TR landing (startDate, sessionName, createdAt)
2. Edit TR > Complete Event > Session List (startDate)
3. TR Landing > View Report (startDate, sessionName, createdAt)

## How to test

See sort specified on TTAHUB-2150.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2150

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

**Comment by thewatermethod** [comment]
do we have a test for this?

**Comment by AdamAdHocTeam** [comment]
> do we have a test for this?

Will add one.

File: src/services/event.ts [hunk 1]
```diff
@@ -98,8 +98,17 @@ async function findEventHelper(where, plural = false): Promise<EventShape | Even
     include: [
       {
         model: SessionReportPilot,
+        attributes: [
+          'id',
+          'eventId',
+          'data',
+          'createdAt',
+          'updatedAt',
+          [sequelize.literal('Date(CASE WHEN "SessionReportPilot".data->>\'startDate\' = \'\' THEN NULL ELSE "SessionReportPilot".data->>\'startDate\' END)'), 'startDate'],
```
Comment (GarrettEHill) [inline src/services/event.ts:107]: Please use backticks and add newlines so it is more readable and then the single quotes don't need to be escaped

File: src/services/event.ts [hunk 2]
```diff
@@ -191,7 +200,16 @@ async function findEventHelperBlob({
...
         as: 'sessionReports',
-        order: [['data.startDate', 'ASC'], ['data.title', 'ASC']],
+        separate: true, // This is required to order the joined table results.
+        attributes: [
+          'id',
+          'eventId',
+          'data',
+          'createdAt',
+          'updatedAt',
+          [sequelize.literal('Date(CASE WHEN "SessionReportPilot".data->>\'startDate\' = \'\' THEN NULL ELSE "SessionReportPilot".data->>\'startDate\' END)'), 'startDate'],
```
Comment (GarrettEHill) [inline src/services/event.ts:210]: Please use backticks and add newlines so it is more readable and then the single quotes don't need to be escaped

File: src/services/event.ts (hunk 1 above)
Comment (GarrettEHill) [reply src/services/event.ts:107]: Also, don't use CASE for this, use NULLIF(<value>, ''), as this is far easier to read and is much faster

File: src/services/event.ts (hunk 2 above)
Comment (GarrettEHill) [reply src/services/event.ts:210]: Also, don't use CASE for this, use NULLIF(<value>, ''), as this is far easier to read and is much faster

File: src/services/sessionReports.ts [hunk 3]
```diff
@@ -37,8 +37,10 @@ async function findSessionHelper(where: WhereOptions, plural = false): Promise<S
       'eventId',
       'data',
       'updatedAt',
+      [sequelize.literal('Date(CASE WHEN "SessionReportPilot".data->>\'startDate\' = \'\' THEN NULL ELSE "SessionReportPilot".data->>\'startDate\' END)'), 'startDate'],
```
Comment (GarrettEHill) [inline src/services/sessionReports.ts:40]: Also, don't use CASE for this, use NULLIF(<value>, ''), as this is far easier to read and is much faster

**Review by thewatermethod — APPROVED** [review]

**Review by kryswisnaskas — APPROVED** [review]
Looks good! 👍

**Pull request merged**